espree.js

```diff
@@ -2560,6 +2560,26 @@ function parseObjectProperty() {
             );
         }
 
+        // destructuring defaults
+        if (allowDestructuring && match("=")) {
+            lex();
+            var value = parseAssignmentExpression();
+            var prop = markerApply(marker, astNodeFactory.createAssignmentExpression("=", id, value));
+            prop.type = astNodeTypes.AssignmentPattern;
+            var shorthand = true; // TODO: figure out the actual value
```

Comment:
kind of unclear why this only works when hard-coded to `true`. We added non-shorthand tests and they seem to pass as well....